nexus/api.go

```diff
@@ -176,3 +203,82 @@ func addContextTimeoutToHTTPHeader(ctx context.Context, httpHeader http.Header)
 	httpHeader.Set(HeaderRequestTimeout, time.Until(deadline).String())
 	return httpHeader
 }
+
+// Link contains an URL and a Type that can be used to decode the URL.
+// Links can contain any arbitrary information as a percent-encoded URL.
+// It can be used to pass information about the caller to the handler, or vice-versa.
+type Link struct {
+	// URL information about the link.
+	// It must be URL percent-encoded.
+	URL string
+	// Type can describe an actual data type for decoding the URL.
+	// Valid chars: alphanumeric, '_', '.', '/'
+	Type string
+}
+
+const linkTypeKey = "type"
+
+func encodeLink(link Link) (string, error) {
+	u, err := url.Parse(link.URL)
+	if err != nil {
+		return "", fmt.Errorf("failed to encode link: url not valid: %s", link.URL)
+	}
+	for _, p := range strings.Split(u.EscapedPath(), "/") {
+		unescaped, err := url.PathUnescape(p)
+		if err != nil || p != url.PathEscape(unescaped) {
+			return "", fmt.Errorf("failed to encode link: path not percent-encoded: %s", link.URL)
+		}
+	}
+	_, err = url.ParseQuery(u.RawQuery)
+	if err != nil {
+		return "", fmt.Errorf("failed to encode link: query not percent-encoded: %s", link.URL)
+	}
+
+	validSymbols := []rune("_./")
+	for _, c := range link.Type {
+		if !unicode.IsLetter(c) && !unicode.IsDigit(c) && !slices.Contains(validSymbols, c) {
```

Comment:
Unicode letters and digits aren't the same as alnum (I think they allow non-ascii chars). Consider:

```suggestion
		if (c < 'a' || c > 'z') && (c < 'A' || c > 'Z') && (c < '0' || c > '9') && c != '_' && c != '.' && c != '/' {
```

Or something like that (didn't test)